<launch>

  <!--Mission specific parameters USV-->
  <arg name="usv_predefined_mission" default="false" />
  <arg name="usv_predefined_mission_name" default="None" />
  <arg name="usv_goal_pose" default="[-74.01067,40.58611,0]" />
  <arg name="usv_initial_pose" default="[-74.001588,40.590419,-3.14]" />

  <!--Monte Carlo Specific Parameters-->
  <arg name="simulation_collection_name" default="colav_crossing_right_v2_"/>
  <arg name="simulations" default="10"/>

  <!--Simulation specific parameters (do not affect the guidance system in any way)-->
  <arg name="vessel_model_file" default="$(find usv_simulator)/config/models/viknes.urdf" />
  <arg name="obstacle_model_file" default="$(find usv_simulator)/config/models/ship1.urdf" />
  <arg name="sim_origin" default="[-74.001588,40.590419,0]" /> 

  <arg name="usv_name" default="Viknes830" />

  <group ns="$(arg usv_name)">
      <param name="predefined_mission" type="bool" value = "$(arg usv_predefined_mission)"/>
      <param name="predefined_mission_name" type="str" value = "$(arg usv_predefined_mission_name)"/>
      <param name="goal_pose" type="yaml" value = "$(arg usv_goal_pose)"/>
      <param name="initial_pose" type="yaml" value = "$(arg usv_initial_pose)"/>

      <param name="simulation_collection_name" type="str" value = "$(arg simulation_collection_name)"/>
      <param name="simulations" type="int" value = "$(arg simulations)"/>

      <param name="sim_origin" type="yaml" value = "$(arg sim_origin)"/>
      <include file="$(find usv_map)/launch/default.launch"/>

      <node pkg="usv_simulator"
            name="sim"
            type="sim_vessel_node"
            respawn="false"
            output="screen">
      </node>
      
      <node pkg="usv_path_trackers"
            name="LOSNode"
            type="los_controller_node.py"
            respawn="false"
            output="screen">
      </node>
      
      <param name="robot_description" textfile="$(arg vessel_model_file)" />
      <node pkg="robot_state_publisher"
          type="robot_state_publisher"
          name="tf_broadcaster" />

      <include file="$(find usv_mission_planner)/launch/mission_planner.launch"/>
      <!--<include file="$(find usv_mission_planner)/launch/mission_planner_client.launch"/>
      -->

      <node pkg="usv_colav"
            name="sb_mpc"
            type="sb_mpc_node"
            respawn="false"
            output="screen">
      </node>
      
      <node pkg="rviz"
        type="rviz"
        name="rviz"
        args="-d $(find usv_simulator)/config/rviz/default.rviz" />

      <node pkg="usv_simulator"
            name="land"
            type="sim_land_node"
            respawn="false"
            output="screen">
      </node>

      <rosparam command="load" file="$(find usv_simulator)/config/monte_carlo_supervisor/leader.yaml" />
      <node pkg="usv_simulator"
            name="monte_carlo_supervisor"
            type="monte_carlo_supervisor_node"
            respawn="false"
            output="screen"
            required="true">
            
      </node>

      <node pkg="usv_simulator"
            name="system_visualization"
            type="sim_system_visualization_node"
            respawn="false"
            output="screen">
      </node>
      
      <include file="$(find usv_realtime_recorder)/launch/recorder.launch"/>
      


  </group>

  
    
</launch>
